### DISCUSSION TITLE
How might one generate an anchored reference to content within an expand element?

### DISCUSSION THREAD

**Discussion by wolfspyre** [thegeeklab/hugo-geekdoc#611] [Ideas]
The desire I had was to enable directly referencing the blob of content encapsulated within an expand section.

If you attempt to add an anchor reference to a heading element within an expand shortcode:

**somepage.md**:
```markdown 
- [Solve all the problems](#winning)
  - [...and then?](#moarwinning)

## Solve all the problems{#winning}
{{< expand "Enjoy Breakfast at Miliways...." "Click to expand">}}
### Already done the impossible? have brunch!{#moarwinning}
Somehow, I feel somewhat less enthused about brunch now...
{{</expand}}
```

Navigating to `foobar.site/somepage#moarwinning`  is odd, as the expand trigger doesn't occur and the user can't see the content... so it felt like a reasonable /expedient way forward to just anchor to the expand element itself.

Looking at the code, I get the sense that it could/might be  possible to add the optional ability to declare the ID of the section, and fallback to autogeneration should the user not explicitly declare it.... 

However, that might not be a great way to accomplish the goal.... I figured that it would make sense to reach out and see if others have way more brilliant ideas of how to address this than I'm coming up with.

TIA! 🐺W

**Comment by xoxys** [comment]
Hi,

To my knowledge, that's just not possible with the JavaScript free approach we use for the expanders. What is the goal anchoring the expander itself? It will still not auto-expand.

**Reply by xoxys** [reply]
If you have important content that needs anchoring, my advice it to no put it into an expand box (as by nature it exists to collapse some additional information to save space).
